kaizen.module

```diff
@@ -1,64 +1,71 @@
-                $file_path = str_replace('COMPONENT', $file_info['dirname'], $file_path);
-                $definitions[$library_name]['css'][$category_name][$file_path] = $file;
-              }
-            }
-          }
-          if (isset($library['js'])) {
-            foreach ($library['js'] as $file_path => $file) {
-              unset($definitions[$library_name]['js'][$file_path]);
-              $file_path = str_replace('COMPONENT', $file_info['dirname'], $file_path);
-              $definitions[$library_name]['js'][$file_path] = $file;
-            }
+  /** @var \Drupal\kaizen\KaizenManager $manager */
+  $manager = \Drupal::service('kaizen.manager');
+
+  $definitions = $manager->getDefinitions();
+
+  foreach ($definitions as $definition) {
+    if ($definition['provider'] != $extension) {
+      continue;
+    }
+    if (isset($definition['library'])) {
+      $library = $definition['library'];
+
+      $relative_path_to_component = str_replace($definition['provider_path'] . '/', "", $definition['component_path']);
+
+      if (isset($library['css'])) {
+        foreach ($library['css'] as $category_name => $category) {
+          foreach ($category as $file_path => $file) {
+            unset($libraries[$definition['id']]['css'][$category_name][$file_path]);
+            $file_path = str_replace('COMPONENT', $relative_path_to_component, $file_path);
+            $libraries[$definition['id']]['css'][$category_name][$file_path] = $file;
           }
         }
-
-        $libraries = array_merge($libraries, $definitions);
+      }
+      if (isset($library['js'])) {
+        foreach ($library['js'] as $file_path => $file) {
+          unset($libraries[$definition['id']]['js'][$file_path]);
+          $file_path = str_replace('COMPONENT', $relative_path_to_component, $file_path);
+          $libraries[$definition['id']]['js'][$file_path] = $file;
+        }
       }
     }
   }
 }
 
+/**
+ * Implements hook_theme().
+ */
 function kaizen_theme($existing, $type, $theme, $path) {
-  $themes = \Drupal::service('theme_handler')->listInfo();
+  /** @var \Drupal\kaizen\KaizenManager $manager */
+  $manager = \Drupal::service('kaizen.manager');
 
+  $definitions = $manager->getDefinitions();
   $templates = [];
 
-  foreach ($themes as $theme) {
-    $theme_path = $theme->getPath();
-    $discovery = new FrontMatterDiscovery([$theme_path], 'formatters', ['plugins', 'formatters'], '/\.frontmatter\.html\.twig$/i');
-    $definitions = $discovery->getDefinitions();
-    foreach ($definitions as $template_name => $template) {
-      // Cut .html.twig file extensiions.
-      $templates[$template_name] = [
-        'template' => substr(pathinfo($template['file'])['basename'], 0, -10),
-        'variables' => [
-          'attributes' => NULL,
-          'content' => NULL,
-        ],
-      ];
-    }
+  foreach ($definitions as $definition) {
+    // Cut .html.twig file extensiions.
+    $templates['kaizen_' . $definition['id']] = [
```

Comment:
I bet you need to convert `-` to `_` as definition could have characters disabled for template names